src/p2p_networking.rs

```diff
@@ -1,25 +1,50 @@
-use crate::errors::RuntimeError;
+use crate::async_runtime::Handle;
+use crate::errors::{LipaError, MapToLipaError, RuntimeError};
 use crate::{NodeAddress, PeerManager};
 use bitcoin::secp256k1::PublicKey;
-use log::debug;
+use log::{debug, error, trace};
+use std::fmt;
 use std::net::SocketAddr;
 use std::str::FromStr;
 use std::sync::Arc;
 use std::task::Poll;
 use std::time::Duration;
+use tokio::task::JoinHandle;
 use tokio::time::sleep;
 
 pub(crate) struct P2pConnections {}
 
 impl P2pConnections {
-    pub async fn connect_peer(
+    pub fn run_bg_connector(
         peer: &NodeAddress,
+        handle1: Handle,
```

Comment:
Maybe rename it to `runtime_handle` or `tokio_handle`?

Reply:
Oh yeah, that was meant to be changed before commiting it :grimacing:

Reply:
Fixed: 440c22abbf62ab577edd2f83b871a1af427ca4ef